lib/modules/ai_bot/bot.rb

```diff
@@ -3,6 +3,25 @@
 module DiscourseAi
   module AiBot
     class Bot
+      class Functions
+        attr_reader :functions
+        attr_reader :current_function
+
+        def initialize
+          @functions = []
+          @current_function = nil
+        end
+
+        def add_function(name)
+          @current_function = { name: name, arguments: +"" }
```

Comment:
What is the `+""` achieving here? I would say nothing given it's an empty string but I might not know something here...

Reply:
He will later mutate the string using `<<`

Reply:
ok I see

```ruby
irb(main):009:0> +"".freeze << "foo"
=> "foo"
irb(main):010:0> "".freeze << "foo"
(irb):10:in `<main>': can't modify frozen String: "" (FrozenError)
	from /Users/joffreyjaffeux/.asdf/installs/ruby/3.2.1/lib/ruby/gems/3.2.0/gems/irb-1.6.3/exe/irb:9:in `<top (required)>'
	from /Users/joffreyjaffeux/.asdf/installs/ruby/3.2.1/bin/irb:25:in `load'
	from /Users/joffreyjaffeux/.asdf/installs/ruby/3.2.1/bin/irb:25:in `<main>'
```

Well first time I see that syntax